### ISSUE TITLE
Understanding the request flow

### ISSUE THREAD

**Issue by rhboyd** [boto/botocore#1702]
When a request is serialized in [Serializer.py](https://github.com/boto/botocore/blob/8a9b18a3632120181eb186ad0685c208160a4a6a/botocore/serialize.py#L80) the request parameters are placed in the body of the request, but some services (such as S3) expect the parameters to be placed in the QueryString. I'm having trouble finding where the translation from the generic request in serializer to the service specific request structure happens. any pointers would be gladly appreciated.

**Comment by JordonPhillips** [comment]
It all happens in the serializer. S3 uses the rest-xml protocol for instance, which splits those all up in [`_partition_parameters`](https://github.com/boto/botocore/blob/8a9b18a3632120181eb186ad0685c208160a4a6a/botocore/serialize.py#L514).

**Comment by rhboyd** [comment]
Thanks Jordon, I think I found it where it's passed into the AWSRequest() constructor and the endpoint dictates how the raw request is transformed into the format that the service expects, is that correct?

**Comment by JordonPhillips** [comment]
The particular serializer the service uses is determined by it's protocol configuration in it's model. [Here](https://github.com/boto/botocore/blob/develop/botocore/data/s3/2006-03-01/service-2.json#L8) is where it's set for S3 for example. Those models contain all the information about how requests for the service are serialized.

**Comment by rhboyd** [comment]
Awesome. I found the service-2.json files last week and was having trouble finding where they were connected from the runtime side. I'm re-suing the botocore data in a cfn macro to make API Gateway integrations a bit easier to build and I was having trouble finding where the generic http request was being translated into the specific type of request each service expected. I ended up pulling the repo and walking through the tests with the debugger to find the translations. Thanks again for your help!
